test/unit/StakingPool/calculatePrice.js

```diff
@@ -1,56 +1,212 @@
+    productType: 1,
+    yieldTokenAddress: AddressZero,
+    coverAssets: 1111,
+    initialPriceRatio: 2000, // 20%
+    capacityReductionRatio: 0,
+    useFixedPrice: false,
+  };
+
+  const productInitializationParams = [
+    {
+      productId: 0,
+      weight: 100, // 1.00
+      initialPrice: coverProductTemplate.initialPriceRatio,
+      targetPrice: 200, // 2%}
+    },
+  ];
+
+  const coverId = 0;
+  const productId = 0;
+
+  beforeEach(async function () {
+    const { stakingPool, cover, nxm, tokenController } = this;
+    const { defaultSender: manager } = this.accounts;
+    const [staker] = this.accounts.members;
+    const productId = 0;
+    // Set global product and product type
+    await cover.setProduct(coverProductTemplate, productId);
+    await cover.setProductType({ claimMethod: 1, gracePeriod: daysToSeconds('7') }, productId);
+
+    // Initialize staking pool
+    const poolId = 0;
+    const isPrivatePool = false;
+    const ipfsDescriptionHash = 'Staking pool 1';
+    const maxPoolFee = 10; // 10%
+    const initialPoolFee = 7; // 7%
+
+    await cover.initializeStaking(
+      stakingPool.address,
+      manager.address,
+      isPrivatePool,
+      initialPoolFee,
+      maxPoolFee,
+      productInitializationParams,
+      poolId,
+      ipfsDescriptionHash,
+    );
+
+    // Deposit into pool
+    const amount = parseEther('50000');
+    await nxm.connect(staker).approve(tokenController.address, amount);
+    await stakingPool.connect(staker).depositTo([
+      {
+        tokenId: 0,
+        amount,
+        destination: staker.address,
+        trancheId: (await getCurrentTrancheId()) + 3,
+      },
+    ]);
   });
 
-  it('should calculate price correctly for active cover below and above surge treshold', async function () {
-    const { stakingPool } = this;
-
-    const amount = parseUnits('700');
-
-    const basePrice = parseUnits('0.026');
-    const activeCover = parseUnits('7800');
-    const capacity = parseUnits('10000');
-
-    const price = await stakingPool.calculatePrice(amount, basePrice, activeCover, capacity);
-
-    const expectedPrice = parseUnits('0.0306');
-
-    // allow for precision error
-    assert.equal(price.div(1e14).toString(), expectedPrice.div(1e14).toString());
-  });
-
-  it('should calculate price correctly for new active cover below surge treshold', async function () {
-    const { stakingPool } = this;
-
-    const amount = parseUnits('1000');
-
-    const basePrice = parseUnits('0.026');
-    const activeCover = parseUnits('1000');
-    const capacity = parseUnits('10000');
-
-    const price = await stakingPool.calculatePrice(amount, basePrice, activeCover, capacity);
-
-    assert.equal(price.toString(), basePrice.toString());
+  async function moveDaysForward(days) {
+    let { timestamp } = await ethers.provider.getBlock('latest');
+    timestamp += daysToSeconds(days.toString());
+    await setNextBlockTime(timestamp);
+    await mineNextBlock();
+  }
+
```

Comment:
as discussed in DMs, let's explore some extreme cases.

Example buying all capacity in 1 go from to 100%.

Anything that has values of 0 or 100% should be explored.